crates/services/producer/src/block_producer.rs

```diff
@@ -171,23 +160,12 @@ where
         height: BlockHeight,
         block_time: Tai64,
     ) -> anyhow::Result<PartialBlockHeader> {
-        let previous_block_info = self.previous_block_info(height)?;
-        let new_da_height = self
-            .select_new_da_height(previous_block_info.da_height)
-            .await?;
+        let mut block_header = self._new_header(height, block_time)?;
+        let new_da_height = self.select_new_da_height(block_header.da_height).await?;
```

Comment:
maybe add some comments around this part of the code, to make it more clear that dry runs should reuse a previous da_height while new blocks should fetch da_height from the relayer. Since this was a big performance issue in the past, I think it's worth making the distinction very clear and explain why.